test/service/test_pagination.py

```diff
@@ -13,6 +12,31 @@ def setUpModule():
     configure_test_logging()
 
 
+internal_pagination_parameters = ('search_after',
+                                  'search_after_uid',
+                                  'search_before',
+                                  'search_before_uid')
+"""
+These pagination parameters are used internally,
+they are should not be included in the `pagination` response field.
+"""
+
+
+def extract_url_params(json_response):
+    """
+    Helper method that extracts parameters for pagination from a response
+    """
+    next_url = json_response['pagination'].get('next', '')
+    previous_url = json_response['pagination'].get('previous', '')
+    return {**parse_url_qs(next_url), **parse_url_qs(previous_url)}
```

Comment:
We can't have this method. If `next` had `&search_before=…&search_after_uid=…` in it and `prev` had `&search_after=…&search_before_uid=…` next and prev would be broken but any test using it would not detect that since it eliminates the separation and merges everything into a single dict, hiding the criss-crossed URL params.